### ISSUE TITLE
Add internal mechanism to broadcast messages via `conda.notices` module

### ISSUE THREAD

**Issue by travishathaway** [conda/conda#13652]
### Checklist

- [X] I added a descriptive title
- [X] I searched open requests and couldn't find a duplicate

### What is the idea?

The notices feature in conda was originally envisioned as being tightly coupled with retrieving notices from channels (either remote or local). But, we could also make this more generic by allowing other parts of the code in conda to broadcast their own notices (e.g. `conda update` notices).

The idea is that other modules would broadcast a notice message that would be collected and then printed out at the end of the command.

### Why is this needed?

The most immediate use case we have for this is a better way to print `conda update` notices for our users. Currently, these update notices are unaware whether they have been displayed before and can be quite repetitive unless the user disables them completely via the `notify_outdated_conda` setting.

### What should happen?

- [ ] Figure out how to broadcast messages that will be displayed later (write these to a file; set-up an internal messaging queue; etc.)
- [ ] Update the `conda.notices` module to read from the above mentioned source to display them
- [ ] Make sure we track what notices have been displayed and which have not

#### Optional (probably better to address in separate issues)
- [ ] Implement this notice system for notifying outdated conda for classic solver
- [ ] Implement this notice system for notifying outdated conda for libmamba solver

**Comment by danpetry** [comment]
I had a use case that might be covered by this if it were expanded to apply to packages?

Full description and discussion [here](https://github.com/conda/conda-libmamba-solver/issues/314), as it's related to the solver, but in summary the use case would be: 

A package variant depends on a virtual package. If the user requests this variant and the platform doesn't support it, currently they get a solver error. I’d like to give them this information in plain english, for less packaging-inclined folks. E.g. for a package depending on `__CUDA >= 11.7`: “Your CUDA version is 11.3, but the CUDA variant of `somepackage` requires drivers that support CUDA 11.7 and above. Please upgrade your system drivers. See https://docs.nvidia.com/deploy/cuda-compatibility/ for more information".

**Comment by travishathaway** [comment]
@danpetry,

This is an applicable use case for this issue.  These messages would need to be broadcasted from the solver. These could be categorized as recommendations that would be shown after the command finishes running.
